src/main/java/essenmakanan/command/EditRecipeCommand.java

```diff
@@ -17,27 +20,122 @@ public EditRecipeCommand(String editDetails, RecipeList recipes) {
+        } catch (EssenInvalidEditException e) {
+            e.handleException();
+        }
+
+        assert !(attributesToEdit==null) : "Attributes is null";
 
-        String recipeName = splitDetails[0];
-        existingRecipe = recipes.getRecipe(recipeName);
+        try {
+            recipes.editRecipe(existingRecipe, attributesToEdit);
+        } catch (EssenFormatException e) {
+            e.handleException();
+        }
 
-        if (existingRecipe == null) {
-            System.out.println("Recipe not found!");
-        } else {
-            try {
-                recipes.editRecipe(existingRecipe, splitDetails);
-            } catch (EssenFormatException e) {
-                e.handleException();
+    }
+
+    /**
+     * Returns the recipe title from the editDetails string.
+     * @return recipe title
+     */
+    public String getRecipeTitle() throws EssenNullInputException {
+        int firstFlag = editDetails.indexOf("/");
+
+        if ( firstFlag < 4 ){
+            // missing other flags or missing title
+            throw new EssenNullInputException();
+        }
+
+        String recipeTitle = editDetails.substring(0, firstFlag-2).trim();
+
+        return recipeTitle;
+
+    }
+
+    /**
+     * Returns the attributes to edit from the editDetails string.
+     * @return attributes to edit
+     */
+    public String[] getAttributesToEdit() throws EssenInvalidEditException {
+        int firstFlag = editDetails.indexOf("/");
+        int totalFlagsMinusTitle = editDetails.split("/").length - 1;
+
+        String[] attributesToEdit = new String[totalFlagsMinusTitle];
+        int attributeCounter = 0;
+
+        while (firstFlag != -1) {
+            if ((firstFlag + 1) >= editDetails.length()) {
+                System.out.println("Please provide details to edit");
+                throw new EssenInvalidEditException();
+            }
+
+            int nextFlag = editDetails.indexOf("/", firstFlag+1);
+
+            if (nextFlag != -1) {
+                attributesToEdit[attributeCounter] = editDetails.substring(firstFlag-1, nextFlag-2).trim();
+            } else {
+                attributesToEdit[attributeCounter] = editDetails.substring(firstFlag-1).trim();
+            }
+            attributeCounter++;
+            firstFlag = nextFlag;
+        }
+
+        if (attributesToEdit.length == 0) {
+            System.out.println("Please provide details to edit");
+            throw new EssenInvalidEditException();
+        }
+
+        for (String attribute : attributesToEdit) {
+            if (!attribute.contains("n/") && !attribute.contains("s/") && !attribute.contains("i/")) {
+                // must have either one of these flags
+                System.out.println("Flag is invalid or format is incorrect.");
+                throw new EssenInvalidEditException();
+            } else if (this.detailsIsNull(attribute)){
+                throw new EssenInvalidEditException();
             }
         }
```

Comment:
Perhaps you should abstract some of the lines. The logic looks good.

Reply:
will do so after functional codes are done